webpack/config.dev.js

```diff
@@ -0,0 +1,14 @@
+var webpack = require('webpack');
+var merge = require('lodash.merge');
+var baseConfig = require('./config.base');
+
+module.exports = merge(baseConfig, {
+  devtool: 'cheap-module-eval-source-map',
+  entry: [
+    'webpack-hot-middleware/client'
+  ].concat(baseConfig.entry),
```

Comment:
@jayphelps If `concat` weren't being used here, it would definitely have unintended side effects, but _with_ `concat`, I believe it should always work as expected. Is there a particular edge case you're thinking of? [This jsbin](https://jsbin.com/felawabeqa/edit?html,js,console) should be helpful in experimenting with `merge`.